controllers/placementapi_controller.go

```diff
@@ -491,19 +492,14 @@ func (r *PlacementAPIReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return result, err
 	}
 
-	// Only expose the service is the deployment succeeded
-	if !instance.Status.Conditions.IsTrue(condition.DeploymentReadyCondition) {
-		Log.Info("Waiting for the Deployment to become Ready before exposing the sevice in Keystone")
-		return ctrl.Result{}, nil
-	}
```

Comment:
I turned out that this check has connection with the check around mariadb account deletion so maybe fixing it in this PR would make sense

Reply:
I added issue and assign it to me https://github.com/openstack-k8s-operators/placement-operator/issues/188 but IMO this is not related at all with current work this is issue introduced during rework of placement on top of tls and later patches. This look like something that was there all the time but starting checking additional states of deployment make it visible. So us a follow up I will readd check and clean flow to correctly apply states in tests

Reply:
Thanks. If it would not be related to this change the we could have kept the condition where it is ;). Anyhow I'm OK to remove it for now and re-introduce it later.

I left comment in #188 about the root cause of test failure we see. It probably needs changes in the mariadb test harness.

Reply:
btw not having the wait for the deployment ready means we are breaking the password rotation as we are allowing the removal of the old mariadbaccount finalizer before our deployment is rolled fully to the new account. But I think we can live with it until the fix for #188